src/theme/ThemeProvider.js

```diff
@@ -0,0 +1,84 @@
+import React, {
+  useState,
+  useLayoutEffect,
+  useMemo,
+  createContext
+} from "react";
+import PropTypes from "prop-types";
+
+export const ThemeContext = createContext();
```

Comment:
Instead of exporting the context itself, is better if you export the `useTheme` so the consumers can easily access the context value (e.g less import statements):

```js
const ThemeContext = createContext();
export const useTheme = () => useContext(ThemeContext);
```